api/v1alpha1/drpolicy_types.go

```diff
@@ -59,3 +92,39 @@ type DRPolicyList struct {
 func init() {
 	SchemeBuilder.Register(&DRPolicy{}, &DRPolicyList{})
 }
+
+// Return a list of unique S3 profiles to upload the relevant cluster state of
+// the given home cluster
+func (d *DRPolicy) S3UploadProfileList(homeCluster string) (s3ProfileList []string) {
+	for _, drCluster := range d.Spec.DRClusterSet {
+		if drCluster.ManagedClusterName != homeCluster {
+			// This drCluster is not the home cluster and is hence, a candidate to
+			// upload cluster state to if this S3 profile is not already on the list.
+			found := false
+
+			for _, s3ProfileName := range s3ProfileList {
```

Comment:
Good question.  The [new DRPolicy API](https://github.com/RamenDR/ramen/pull/117/files#diff-4087c0057f96eb96ad38697c97c920a27e4741ddfa6c64cf4a5a3f649f223b20R62) in this PR moves the [S3ProfileName ](https://github.com/RamenDR/ramen/pull/117/files#diff-4087c0057f96eb96ad38697c97c920a27e4741ddfa6c64cf4a5a3f649f223b20R53) as a per-cluster mandatory field DRPolicy.spec.DRClusterSet[n].s3ProfileName, to handle the case where an s3 store is located in each of the managed clusters.  On the other hand, in the case of a single common S3 store (in the hub or in the cloud), this interface requires specifying the same s3ProfileName (e.g., s3-in-hub) for both the managed clusters.

An alternate API design that I considered defines an optional DRPolicy.spec.s3ProfileName for the case of the single common S3 store, in addition to the DRPolicy.spec.DRClusterSet[n].s3ProfileName.  This approach requires making both fields optional and I was worried that this will make it error prone & lead to confusion and, hence, did not choose this alternate design.